torch/csrc/autograd/profiler_python.cpp

```diff
@@ -868,6 +892,18 @@ class PostProcess {
     }
   }
 
+  void set_start_frames(
+      const std::vector<PythonTracer::StartFrame>& start_frames,
+      std::vector<python_tracer::CompressedEvent>& enters) {
+    for (const auto& frame : start_frames) {
+      enters.push_back(
+          {frame.trace_key_,
+           NoTID, // Allows us to detect unhandled start frames
+           enters.begin()->kineto_info_,
```

Comment:
While testing this on a pretty big job the crashes appear consistently on this line. `enters` is empty, though I can't repro it in small experiments.

In any case, this logic appears flawed. `kineto_info_` contains `thread_id` so it can't be set equal for events. We should do the same patching logic as for the thread id above to populate it. It seems that there's `Result::kineto_info_` preserved intact.

Side question: isn't the thread id in `kineto_info_` and at the top level the same? Why do we need both?

cc @jamesr66a

Reply:
Following up on this bug (independent of the GIL), it can happen that `enters` is empty, and then it'd crash in post-processing. Do you agree with my reasoning above?

Reply:
Yeah, this seems wrong. Both because of the crash and because it's logically incorrect. I have no idea how that got there. (Maybe it was meant to be a placeholder too?)

The crash happens if there are enter events, but no python executes in the profiled region. This is impossible with the context manager since it has to run `__exit__` at least, but not so for on-demand. (A simple repro is just to trace `sleep`.) So your workload must be doing something else (long C++ kernel, blocked waiting on dataloader, etc.) I'm just going to make it default construct for now since I don't think it's particularly load bearing. (And we can revisit if or when it matters.)

As for why TID is stored multiple places: it's possible to build profiler without Kineto. That means that Profiler can't use any Kineto structs in its core logic because if it does the `#ifdef`s will blow up the non-kineto build. (We always include one kineto header to get an enum, but even that was very painful.) So until there is any consolidation profiler handles its own logic and kineto metadata is mostly along for the ride.